src/structured_logging/log_config.py

```diff
@@ -0,0 +1,300 @@
+            "propagate": False,
+        },
+        "airflow.task": {
+            "handlers": ["task"],
+            "level": LOG_LEVEL,
+            "propagate": False,
+            "filters": ["mask_secrets"],
+        },
+        "flask_appbuilder": {
+            "handlers": ["console"],
+            "level": FAB_LOG_LEVEL,
+            "propagate": True,
+        },
+    },
+    "root": {
+        "handlers": ["console"],
+        "level": LOG_LEVEL,
+        "filters": ["mask_secrets"],
+    },
+}
+
+EXTRA_LOGGER_NAMES: str = conf.get("logging", "EXTRA_LOGGER_NAMES", fallback=None)
+if EXTRA_LOGGER_NAMES:
+    new_loggers = {
+        logger_name.strip(): {
+            "handlers": ["console"],
+            "level": LOG_LEVEL,
+            "propagate": True,
+        }
+        for logger_name in EXTRA_LOGGER_NAMES.split(",")
+    }
+    LOGGING_CONFIG["loggers"].update(new_loggers)
+
+DEFAULT_DAG_PARSING_LOGGING_CONFIG: dict[str, dict[str, dict[str, Any]]] = {
+    "handlers": {
+        "processor_manager": {
+            # CUSTOM: added reference to the custom log handler class.
+            "class": "structured_logging.logging_handler.JsonRotatingFileHandler",
+            # 'formatter': 'airflow',
+            "filename": DAG_PROCESSOR_MANAGER_LOG_LOCATION,
+            "mode": "a",
+            "maxBytes": 104857600,  # 100MB
+            "backupCount": 5,
+        }
+    },
+    "loggers": {
+        "airflow.processor_manager": {
+            "handlers": ["processor_manager"],
+            "level": LOG_LEVEL,
+            "propagate": False,
+        }
+    },
+}
+
+# Only update the handlers and loggers when CONFIG_PROCESSOR_MANAGER_LOGGER is set.
+# This is to avoid exceptions when initializing RotatingFileHandler multiple times
+# in multiple processes.
+if os.environ.get("CONFIG_PROCESSOR_MANAGER_LOGGER") == "True":
+    LOGGING_CONFIG["handlers"].update(DEFAULT_DAG_PARSING_LOGGING_CONFIG["handlers"])
+    LOGGING_CONFIG["loggers"].update(DEFAULT_DAG_PARSING_LOGGING_CONFIG["loggers"])
+
+    # Manually create log directory for processor_manager handler as RotatingFileHandler
+    # will only create file but not the directory.
+    processor_manager_handler_config: dict[str, Any] = DEFAULT_DAG_PARSING_LOGGING_CONFIG[
+        "handlers"
+    ]["processor_manager"]
+    directory: str = os.path.dirname(processor_manager_handler_config["filename"])
+    Path(directory).mkdir(parents=True, exist_ok=True, mode=0o755)
+
+##################
+# Remote logging #
+##################
+
+REMOTE_LOGGING: bool = conf.getboolean("logging", "remote_logging")
+
+if REMOTE_LOGGING:
+
+    ELASTICSEARCH_HOST: str = conf.get("elasticsearch", "HOST")
+
+    # Storage bucket URL for remote logging
+    # S3 buckets should start with "s3://"
+    # Cloudwatch log groups should start with "cloudwatch://"
+    # GCS buckets should start with "gs://"
+    # WASB buckets should start with "wasb"
+    # just to help Airflow select correct handler
```

Comment:
We will use only `wasb`. The other options mentioned in the comments and in the code blocks below are unnecessary. Maybe we could remove some code? Or if this is a file that has been plucked from the inter webs, leave it as is so that we can more easily copy updates in the future.

Reply:
Yes this is the official Airflow template that is then overwritten by the custom log handlers references. So basically  the rest needs to be untouched as Airflow is expecting it